src/main/kotlin/io/libp2p/security/noise/NoiseXXSecureChannel.kt

```diff
@@ -0,0 +1,315 @@
+        return initChannel(ch, "")
+    }
+
+    override fun initChannel(
+            ch: P2PAbstractChannel,
+            selectedProtocol: String
+    ): CompletableFuture<SecureChannel.Session> {
+        role = if (ch.isInitiator) AtomicInteger(HandshakeState.INITIATOR) else AtomicInteger(HandshakeState.RESPONDER)
+
+        chid = "ch=" + ch.nettyChannel.id().asShortText() + "-" + ch.nettyChannel.localAddress() + "-" + ch.nettyChannel.remoteAddress()
+        logger.debug(chid)
+
+        val ret = CompletableFuture<SecureChannel.Session>()
+        val resultHandler = object : ChannelInboundHandlerAdapter() {
+            override fun userEventTriggered(ctx: ChannelHandlerContext, evt: Any) {
+                when (evt) {
+                    is SecureChannelInitialized -> {
+                        val session = evt.session as NoiseSecureChannelSession
+                        ctx.channel().attr(SECURE_SESSION).set(session)
+
+                        ret.complete(session)
+                        ctx.pipeline().remove(this)
+                        ctx.pipeline().addLast(NoiseXXCodec(session.aliceCipher, session.bobCipher))
+
+                        logger.debug("Reporting secure channel initialized")
+                    }
+                    is SecureChannelFailed -> {
+                        ret.completeExceptionally(evt.exception)
+
+                        ctx.pipeline().remove(handshakeHandlerName + chid)
+                        ctx.pipeline().remove(this)
+
+                        logger.debug("Reporting secure channel failed")
+                    }
+                }
+                ctx.fireUserEventTriggered(evt)
+                ctx.fireChannelActive()
+            }
+        }
+        ch.nettyChannel.pipeline().addLast(handshakeHandlerName + chid, NoiseIoHandshake())
+        ch.nettyChannel.pipeline().addLast(handshakeHandlerName + chid + "ResultHandler", resultHandler)
+        return ret
+    }
+
+    inner class NoiseIoHandshake() : SimpleChannelInboundHandler<ByteBuf>() {
+
+        private val handshakestate: HandshakeState = HandshakeState(protocolName, role.get())
+        private var loggerName: String
+        private var logger2: Logger
+
+        private var localNoiseState: DHState
+        private var sentNoiseKeyPayload = false
+
+        private val instancePayload = ByteArray(65535)
+        private var instancePayloadLength = 0
+
+        init {
+            val roleString = if (role.get() == HandshakeState.INITIATOR) "INIT" else "RESP"
+//            loggerName = loggerNameParent + "." + this.hashCode().toString().substring(5)
+            loggerName = roleString + "|" + chid + "|" + loggerNameParent
+            loggerName = loggerName.replace(".", "_")
+//            System.out.println("loggerName:"+loggerName)
+
+            logger2 = LogManager.getLogger(loggerName)
+            Configurator.setLevel(loggerName, Level.DEBUG)
```

Comment:
Why do we need these logger tweaks here? Also the programmatic level (hardcoded) configuration isn't good

Reply:
Done.